packages/fxa-settings/src/lib/sensitive-data-client.ts

```diff
@@ -4,8 +4,69 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+export namespace SensitiveData {
+  /**
+   * The keys for the various kinds of data that can be inserted into a `SensitiveData` object.
+   *
+   * @see SensitiveDataClient for implementations.
+   */
+  export enum Key {
+    Auth = 'auth',
+    AccountReset = 'accountResetData',
+    NewRecoveryKey = 'newRecoveryKeyData',
+  }
+
+  /**
+   * A data type for the various data added through the {@link SensitiveDataClient}.
+   *
+   * N.B. Use a non-nullable type if the values are derived from a hook during initialization.
+   *
+   * @see SensitiveDataClient for implementations.
+   */
+  export type DataMap = {
```

Comment:
So nice having all these types set up! \o/ Can we go ahead and get rid of `SensitiveDataClientAuthKeys` / `SensitiveDataClientData` with these changes?

Reply:
I've added this as a third patch in this stack to make it easier for reviewing. This change touches _a lot_ of files and I think we can also land this separately if fixing the tests prove difficult (will check them out tomorrow).